packages/build/src/buildbot_client/main.js

```diff
@@ -0,0 +1,76 @@
+const net = require('net')
+
+const { addErrorInfo } = require('../error/info')
+
+const startBuildbotClient = function(buildbotServerSocket) {
+  return new Promise((resolve, reject) => {
+    try {
+      const client = net.createConnection(buildbotServerSocket)
+      client.on('connect', () => resolve(client))
+    } catch (error) {
+      addErrorInfo(error, {
+        type: 'buildbotClientStartup',
+        location: { buildbotServerSocket },
+      })
+      reject(error)
+    }
+  })
+}
+
+const closeBuildbotClient = function(client) {
+  client.end()
+  return
+}
+
+const writeDeployPayload = function(client, payload) {
+  return new Promise((resolve, reject) => {
+    try {
+      client.write(JSON.stringify(payload), resolve)
+    } catch (e) {
+      reject(e)
```

Comment:
Should this error have more context?

```suggestion
          const error = `Could not send TCP payload to the buildbot: ${e.message}\n${JSON.stringify(payload, null, 2)}`
          reject(error)
```